### PR TITLE
Support for chain pattern addFieldValidation method.

### PR DESCRIPTION
**Pull request by crsanti**
This resolves issue #27.
PD: I've indented the spec file with 2 spaces.

### CODE REVIEW

**Comment by brauliodiez** [comment]
@Nasdan , @JaimeSalas  this is related to this issue:

https://github.com/Lemoncode/lcFormValidation/issues/27

Note down that @crsanti has updated the tab spaces, some test may look as removed but I guess is the viewer that has made some mess

File: lib/src/spec/validationEngine.spec.ts [hunk 1]
```diff
@@ -6,168 +6,198 @@ import { FieldValidationResult } from '../entities';
...
+        done();
+      });
+  });
+
 
-    it('should return isFormDirty true after initialization and set value field',(done) => {
-        // Arrange
-        const formValidationBase : ValidationEngine = new ValidationEngine();
-        const viewModel = [{formFieldName: 'nameId', vmFieldName: 'name'}];
+  it('should return isFormDirty false after initiazalition', () => {
```
Comment (nasdan) [inline lib/src/spec/validationEngine.spec.ts:38]: typo: initialization

File: lib/src/spec/validationEngine.spec.ts [hunk 2]
```diff
@@ -6,168 +6,198 @@ import { FieldValidationResult } from '../entities';
```
Comment (nasdan) [inline lib/src/spec/validationEngine.spec.ts:6]: Is this file updated by tabs spaces only?

**Review by nasdan — COMMENTED** [review]
Pending to add specs testing chain functionality, I mean add specs using chain pattern in `validationEngineValidateSingleField.spec.ts` and `validationEngineValidateForm.spec.ts`

File: lib/src/spec/validationEngine.spec.ts (hunk 2 above)
Comment (nasdan) [reply lib/src/spec/validationEngine.spec.ts:6]: It isn't. I saw last [specs](https://github.com/Lemoncode/lcFormValidation/pull/32/files#diff-0cca4b84b6f407ccfec363df7bfcf4c8R171)

File: lib/src/spec/validationEngine.spec.ts [hunk 3]
```diff
@@ -6,168 +6,198 @@ import { FieldValidationResult } from '../entities';
 
 //TODO: Implement Issue #20
```
Comment (nasdan) [inline lib/src/spec/validationEngine.spec.ts:7]: I think this issue number comes from bitbucket private repository. Maybe we need new issue to update TODOs

File: lib/src/spec/validationEngine.spec.ts [hunk 4]
```diff
@@ -6,168 +6,198 @@ import { FieldValidationResult } from '../entities';
...
+    let result = validationEngine.isFieldKeyMappingDefined(key);
+
+    //Assert
+    expect(result).to.be.false;
+  });
+
+  it('should returns isFieldKeyMappingDefined equals true ' +
+      'when passing a key equals "test" and validationsPerField equals ["test": validationFn]', () => {
+    //Arrange
+    let key = "test";
```
Comment (JaimeSalas) [inline lib/src/spec/validationEngine.spec.ts:157]: It could be const instead let in variable key?

File: lib/src/spec/validationEngine.spec.ts [hunk 5]
```diff
@@ -6,168 +6,198 @@ import { FieldValidationResult } from '../entities';
...
+    let validationFn = (vm, value) => {
```
Comment (JaimeSalas) [inline lib/src/spec/validationEngine.spec.ts:158]: Also with validationFn?

File: lib/src/spec/validationEngine.spec.ts [hunk 6]
```diff
@@ -6,168 +6,198 @@ import { FieldValidationResult } from '../entities';
...
+        let fieldValidationResult = new FieldValidationResult();
+        return Promise.resolve(fieldValidationResult);
+    }
+    let validationEngine = new ValidationEngine();
+    validationEngine.addFieldValidationAsync(key, validationFn);
```
Comment (JaimeSalas) [inline lib/src/spec/validationEngine.spec.ts:163]: And maybe validationEngine, could be const as well?

File: lib/src/spec/validationEngine.spec.ts (hunk 4 above)
Comment (crsanti) [reply lib/src/spec/validationEngine.spec.ts:157]: Sure!, i've only reformatted and added the two last tests, though. Should I change previous tests??

File: lib/src/spec/validationEngine.spec.ts (hunk 1 above)
Comment (JaimeSalas) [reply lib/src/spec/validationEngine.spec.ts:38]: Look at all the test suite, I think that variables are not consistent (let / const) declare.

**Comment by brauliodiez** [comment]
O think for next times rather Split in two cases, one is tab formatting, the other implementing the case. This case is polluted with all the tab stuff, is Hard to review

File: lib/src/spec/validationEngine.spec.ts (hunk 4 above)
Comment (JaimeSalas) [reply lib/src/spec/validationEngine.spec.ts:157]: Ok, then is not on scope. That has to be done on separate issue. On my point of view, you can go ahead with merge.

**Pull request merged**